### ISSUE TITLE
Class Scanning

### ISSUE THREAD

**Issue by evilband7** [Roave/BetterReflection#213]
I need an example for scanning all classes in specific directories and resulted as ClassRefelction[]

**Comment by asgrim** [comment]
You'll want to write a custom `SourceLocator` to scan the directory. You may be able to decorate the `SingleFileSourceLocator`, but you'd need to check how that works.

You can then use your custom source locator like:

```php
use BetterReflection\Reflector\ClassReflector;

$reflector = new ClassReflector(new DirectoryIteratingSourceLocator($classLoader));
$classReflections = $reflector->getAllClasses();
```

The interface your `SourceLocator` needs to implement is [here](https://github.com/Roave/BetterReflection/blob/master/src/SourceLocator/Type/SourceLocator.php), particularly of interest here for you is the `locateIdentifiersByType` method.

Note; if you are able to implement this and provide comprehensive tests & documentation, we would likely accept a pull request to add this as a new source locator in the library :)

Hope this helps, thanks!

**Comment by evilband7** [comment]
Thanks. I'll try to make a pull request. xD

**Comment by asgrim** [comment]
This is resolved by #214 now :)
